### DISCUSSION TITLE
Instructions from README.md not working

### DISCUSSION THREAD

**Discussion by jrddupont** [khoih-prog/TimerInterrupt#21] [Q&A] [answered]
I am trying to use a TimerInterrupt on Timer1 on an Arduino Uno R3 using this method: https://github.com/khoih-prog/TimerInterrupt#11-init-hardware-timer  
So I have this source code:  

```
#include <TimerInterrupt.h>
#include <ISR_Timer.h>

#define USE_TIMER_1     true

ITimer1.init();
```  
  
But I immediately get this error:   
```
TimerTest:6:1: error: 'ITimer1' does not name a type; did you mean 'ISR_Timer'?
 ITimer1.init();
 ^~~~~~~
 ISR_Timer
exit status 1
'ITimer1' does not name a type; did you mean 'ISR_Timer'?
```  
  
Am I following the instructions wrong? According to the source code, that object should exist: https://github.com/khoih-prog/TimerInterrupt/blob/master/src/TimerInterrupt.h#L477

**Comment by khoih-prog** [comment, marked as answer]
Thanks for your interest of the library.

You have to declare `#define USE_TIMER_1     true` before the `#include <TimerInterrupt.h>`, as follows

```
#define USE_TIMER_1     true

#include <TimerInterrupt.h>
#include <ISR_Timer.h>

void setup()
{
  ITimer1.init();
}

void loop() 
{
```

As usual, it's always better to start with a simplest example to have preliminary feelings and understand the basic operation of the library.

Good Luck,

**Reply by jrddupont** [reply]
Thank you! That makes a lot of sense 😅
